packages/react-aria-components/docs/Button.mdx

```diff
@@ -213,8 +213,71 @@ A `Button` can be disabled using the `isDisabled` prop.
 <Button isDisabled>Pin</Button>
 ```
 
+
 ## Advanced customization
 
+### Contexts
+
+All React Aria Components export a corresponding context that can be used to send props to them from a parent element. This enables you to build your own compositional APIs similar to those found in React Aria Components itself. You can send any prop via context that you could pass to the corresponding component, and refs are supported as well. The local props and ref on the component are merged with the ones passed via context, with the local props taking precedence (following the rules documented in [mergeProps](mergeProps.html)).
+
+<ContextTable components={['Button']} docs={docs} />
+
+This example shows a `ButtonGroup` component that renders a group of buttons. The entire group can be marked as disabled via the `isDisabled` prop, which is passed to all child buttons via the `ButtonContext` provider.
+
+```tsx example
+import {ButtonContext} from 'react-aria-components';
+
+interface ButtonGroupProps {
+  children?: React.ReactNode,
+  isDisabled?: boolean
+}
+
+function ButtonGroup({children, isDisabled}: ButtonGroupProps) {
+  return (
+    <div style={{display: 'flex', gap: 8}}>
+      {/*- begin highlight -*/}
+      <ButtonContext.Provider value={{isDisabled}}>
+      {/*- end highlight -*/}
+        {children}
+      </ButtonContext.Provider>
+    </div>
+  );
+}
+
+<ButtonGroup isDisabled>
+  <Button>Save</Button>
+  <Button>Publish</Button>
+</ButtonGroup>
+```
+
 ### Hooks
 
-If you need to customize things further, such as accessing internal state or customizing DOM structure, you can drop down to the lower level Hook-based API. See [useButton](useButton.html) for more details.
+If you need to customize things further, such as intercepting events or customizing DOM elements, you can drop down to the lower level Hook-based API. Consume from `ButtonContext` in your component with <TypeLink links={docs.links} type={docs.exports.useContextProps} /> to make it compatible with other React Aria Components. See [useButton](useButton.html) for more details.
+
+This example uses [Framer Motion](https://www.framer.com/motion/) to create an `AnimatedButton` component that animates based on the `isPressed` state provided by `useButton`. It can be used standalone or as a part of any React Aria component.
```

Comment:
Yeah exactly, I didn't want to actually import `framer-motion` in the docs because it is somewhat large and this is just showing a technique. Should be easy enough to copy paste if you want